packages/@apphosting/adapter-nextjs/src/overrides.ts

```diff
@@ -158,38 +161,47 @@ export async function addRouteOverrides(
   distDir: string,
   adapterMetadata: AdapterMetadata,
 ) {
-  const middlewareManifest = loadMiddlewareManifest(appPath, distDir);
   const routeManifest = loadRouteManifest(appPath, distDir);
+
+  // Add the adapter version to all routes
   routeManifest.headers.push({
     source: "/:path*",
     headers: [
       {
         key: "x-fah-adapter",
         value: `nextjs-${adapterMetadata.adapterVersion}`,
       },
-      ...(middlewareExists(middlewareManifest)
-        ? [
-            {
-              key: "x-fah-middleware",
-              value: "true",
-            },
-          ]
-        : []),
     ],
     /*
-      NextJs converts the source string to a regex using path-to-regexp (https://github.com/pillarjs/path-to-regexp) at
-      build time: https://github.com/vercel/next.js/blob/canary/packages/next/src/build/index.ts#L1273.
-      This regex is then used to match the route against the request path.
+    NextJs converts the source string to a regex using path-to-regexp (https://github.com/pillarjs/path-to-regexp) at
+    build time: https://github.com/vercel/next.js/blob/canary/packages/next/src/build/index.ts#L1273.
+    This regex is then used to match the route against the request path.
 
-      This regex was generated by building a sample NextJs app with the source string `/:path*` and then inspecting the
-      routes-manifest.json file.
-    */
+    This regex was generated by building a sample NextJs app with the source string `/:path*` and then inspecting the
+    routes-manifest.json file.
+  */
     regex: "^(?:/((?:[^/]+?)(?:/(?:[^/]+?))*))?(?:/)?$",
   });
 
-  await writeRouteManifest(appPath, distDir, routeManifest);
-}
+  // Add the middleware header to all routes for which middleware is enabled
+  const middlewareManifest = loadMiddlewareManifest(appPath, distDir);
+  const rootMiddleware = middlewareManifest.middleware["/"];
```

Comment:
@jamesdaniels My assumption here was that we only need to look at the root key as NextJs now only supports middleware at the root directory. It used to support nested middleware while in beta which is why I think the API is like this.

Similar logic in Opennext AWS adapter: https://github.com/opennextjs/opennextjs-aws/blob/73281c958d43d865fdded0d5d86dd82747365fee/packages/open-next/src/core/routing/util.ts#L160-L174

Reply:
Yeah that's a good assumption